data_structures/stack_using_two_queues.cpp

```diff
@@ -0,0 +1,245 @@
+/**
+ * @file
+ * @brief Implementation of stack using two queues
+ * @author [Gopal-Dahale](https://github.com/Gopal-Dahale)
+ * @details
+ * In this alogrithm, stack is implemented by making the pop operation costly.
+ * push(x) operation enqueues x to q1.
+ * pop() operation puts the first n-1 elements from q1 to q2 and then dequeue
+ * the last element from q1. Finally the names q1 and q2 are swapped. The time
+ * Complexity of the algorithm is \f$O(n)\f$ for pop() and \f$O(1)\f$ for
+ * push().
+ */
+
+#include <cassert>
+#include <iostream>
+#include <queue>
+#include <string>
+
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures
+{
+    template <typename T>
+    class Stack
+    {
+    private:
+        std::queue<T> q1, q2;
+
+    public:
+        /**
+         * Pushes the given element to top of stack
+         */
+        void push(T data) { q1.push(data); }
+
+        /**
+         * Pops the topmost element of stack
+         */
+        void pop()
+        {
+            // push first n-1 elements from q1 to q2
+            // pop the last element from q1
+            // swap q1 and q2
+            if (q1.size() == 0)
+            {
+                return;
+            }
+
+            while (q1.size() > 1)
+            {
+                q2.push(q1.front());
+                q1.pop();
+            }
+            // pops the last element from q1;
+            q1.pop();
+
+            // swaps q1 with q2
+            q1.swap(q2);
+        }
+
+        /**
+         * Returns the topmost element of stack.
+         * Returns -1 if stack is empty
+         */
+        T top()
+        {
+            if (q1.size() == 0)
+            {
+                return T();
+            }
+
+            while (q1.size() > 1)
+            {
+                q2.push(q1.front());
+                q1.pop();
+            }
+
+            T top_element = q1.front();
+            q2.push(q1.front());
+            q1.pop();
+
+            // swaps q1 with q2
+            q1.swap(q2);
+
+            return top_element;
+        }
+
+        /**
+         * Returns the size of stack
+         */
+        int size() { return q1.size() + q2.size(); }
+
+        /**
+         * Returns true if stack is empty, otherwise false
+         */
+        bool empty() { return (q1.size() == 0); }
+    };
+
+} // namespace data_structures
+
+/**
```

Comment:
```suggestion
namespace { 
/**
```
please use an un-named namespace for classes and variables that are only used in this program and used to show as examples